packages/collector/test/tracing/cloud/aws/lambda/test.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * (c) Copyright IBM Corp. 2021
+ * (c) Copyright Instana Inc. and contributors 2021
+ */
+
+'use strict';
+
+// const semver = require('semver');
+const path = require('path');
+const { expect } = require('chai');
+const { fail } = expect;
+const constants = require('@instana/core').tracing.constants;
+const supportedVersion = require('@instana/core').tracing.supportedVersion;
+const config = require('../../../../../../core/test/config');
+const { expectExactlyOneMatching, retry, stringifyItems, delay } = require('../../../../../../core/test/test_util');
+const ProcessControls = require('../../../../test_util/ProcessControls');
+const globalAgent = require('../../../../globalAgent');
+
+const SPAN_NAME = 'aws.lambda.invoke';
+const functionName = 'team-nodejs-invoke-function';
+let mochaSuiteFn;
+
+const withErrorOptions = [false, true];
+const requestMethods = ['Callback', 'Promise'];
+const availableOperations = ['invoke', 'invokeAsync'];
+
+if (!supportedVersion(process.versions.node)) {
+  mochaSuiteFn = describe.skip;
+} else {
+  mochaSuiteFn = describe;
+}
+
+const retryTime = config.getTestTimeout() * 2;
+
+mochaSuiteFn('tracing/cloud/aws/lambda', function () {
+  this.timeout(config.getTestTimeout() * 3);
+  globalAgent.setUpCleanUpHooks();
+  const agentControls = globalAgent.instance;
+  describe('tracing enabled, no suppression', function () {
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      env: {
+        AWS_LAMBDA_FUNCTION_NAME: functionName
+      }
+    });
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
+    withErrorOptions.forEach(withError => {
+      requestMethods.forEach(requestMethod => {
+        describe(`getting result with: ${requestMethod}; with error: ${withError ? 'yes' : 'no'}`, () => {
+          availableOperations.forEach(operation => {
+            it(`operation: ${operation}`, async () => {
+              const withErrorOption = withError ? '?withError=1' : '';
+              const apiPath = `/${operation}/${requestMethod}`;
+              const response = await appControls.sendRequest({
+                method: 'GET',
+                path: `${apiPath}${withErrorOption}`,
+                simple: withError === false
+              });
+              return verify(appControls, response, apiPath, operation, withError);
+            });
+          });
+        });
+      });
+    });
+    function verify(controls, response, apiPath, operation, withError) {
+      return retry(() => {
+        return agentControls.getSpans().then(spans => verifySpans(controls, spans, apiPath, operation, withError));
```

Comment:
Suggestion: Since we manipulate the client context in our instrumentation, it would be great if we somehow could verify in the test that that content the test application put in there is still in there after our instrumentation manipulated it. I'm not sure how to achieve that - maybe if we add the client context to the http response after executing the lambda invocation (see above).